ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/container/v2/container/clientlibs/site/js/formcontainerview.js

```diff
@@ -72,6 +72,29 @@
                     window.alert("Issue while saving draft");
                 }
             }, "saveError");
+            this.#setupAutoSave(self.getModel());
+        }
+
+        /**
+         * Register time based auto save
+         * @param formModel.
+         */
+         #setupAutoSave(formModel) {
+            const autoSaveProperties = formModel?.properties?.['fd:autoSave'];
+            const enableAutoSave = autoSaveProperties?.['fd:enableAutoSave'];
+            if (enableAutoSave) {
+                const autoSaveStrategyType = autoSaveProperties['fd:autoSaveStrategyType'];
+                const autoSaveInterval = autoSaveProperties['fd:autoSaveInterval'];
+                const saveEndPoint = FormView.Utils.getContextPath() + '/adobe/forms/af/save/' + formModel.id;
+                if (autoSaveStrategyType === 'time' && autoSaveInterval) {
+                    console.log("Registering time based auto save");
+                    setInterval(() => {
+                        formModel.dispatch(new FormView.Actions.Save({
+                            'action': saveEndPoint
```

Comment:
using the term action to refer to an endpoint URL is somewhat unconventional, and it could lead to confusion

Reply:
Save event calls Form model save method which internal calls submit method with save metadata.
Submit method uses action to fetch end point that's why here we need to specify action for end point.
@rismehta

Reply:
I understand this, but the payload looks confusing and needs to be changed. Let's make the payload explicit signifying thats this is a saveEndPoint